# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../jbig2enc/gcc4.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- jbig2enc-0.23/jbig2segments.h.vanilla	2008-10-28 16:45:45.000000000 +0100
+++ jbig2enc-0.23/jbig2segments.h	2008-10-28 16:45:51.000000000 +0100
@@ -4,6 +4,7 @@
 #ifndef THIRD_PARTY_JBIG2ENC_JBIG2SEGMENTS_H__
 #define THIRD_PARTY_JBIG2ENC_JBIG2SEGMENTS_H__
 
+#include <string.h>
 #include <vector>
 #include <netinet/in.h>
 
--- jbig2enc-0.23/jbig2enc.cc.vanilla	2008-10-28 16:46:04.000000000 +0100
+++ jbig2enc-0.23/jbig2enc.cc	2008-10-28 16:50:24.000000000 +0100
@@ -19,6 +19,7 @@
 #include <vector>
 #include <algorithm>
 
+#include <stdio.h> // for the leptonlib allheader.h due g++
 #include <allheaders.h>
 #include <pix.h>
 
--- jbig2enc-0.27/jbig2sym.cc.vanilla	2008-10-28 16:54:58.000000000 +0100
+++ jbig2enc-0.27/jbig2sym.cc	2008-10-28 16:55:02.000000000 +0100
@@ -18,6 +18,8 @@
 #include <map>
 #include <algorithm>
 
+#include <stdio.h>
+
 #include "jbig2arith.h"
 
 #ifdef _MSC_VER
--- jbig2enc-0.27/jbig2.cc.vanilla	2008-10-28 16:55:15.000000000 +0100
+++ jbig2enc-0.27/jbig2.cc	2008-10-28 16:56:22.000000000 +0100
@@ -21,6 +21,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
 
 #include <allheaders.h>
 #include <pix.h>
--- jbig2enc-0.27/jbig2arith.cc.vanilla	2008-10-28 16:53:52.000000000 +0100
+++ jbig2enc-0.27/jbig2arith.cc	2008-10-28 16:54:13.000000000 +0100
@@ -16,6 +16,8 @@
 // limitations under the License.
 
 #include "jbig2arith.h"
+
+#include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
 
